src/PolykeyAgent.ts

```diff
@@ -573,6 +573,7 @@ class PolykeyAgent {
         PolykeyAgent.eventSymbols.Proxy,
         async (data: ConnectionData) => {
           if (data.type === 'reverse') {
+            if (this.keyManager.getNodeId().equals(data.remoteNodeId)) return;
```

Comment:
Do we still need this if the filtering of our own NodeID is all done in the NCM now? Is this doing double filtering? If so, we shouldn't need this.

Reply:
Yes and no, here we're doing it to avoid trying to add it to the node graph when we receive a connection. The `NodeGraph` already checks this so functionally we're just not printing the log message for it.

Reply:
Please add a comment about how this is already being filtered out, but it's done here earlier to avoid posting another log message.

It seems not exactly tidy atm... but this will involve thinking about #444